contracts/messaging/GasFeePricingUpgradeable.sol

```diff
@@ -0,0 +1,433 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import "./framework/SynMessagingReceiverUpgradeable.sol";
+import "./interfaces/IGasFeePricing.sol";
+import "./libraries/Options.sol";
+
+contract GasFeePricingUpgradeable is SynMessagingReceiverUpgradeable, IGasFeePricing {
+    /*┌──────────────────────────────────────────────────────────────────────┐
+      │                               STRUCTS                                │
+      └──────────────────────────────────────────────────────────────────────┘*/
+
+    /// @dev Dst chain's basic variables, that are unlikely to change over time.
+    struct ChainConfig {
+        // Amount of gas units needed to receive "update chainInfo" message
+        uint128 gasAmountNeeded;
+        // Maximum gas airdrop available on chain
+        uint128 maxGasDrop;
+    }
+
+    /// @dev Information about dst chain's gas price, which can change over time
+    /// due to gas token price movement, or gas spikes.
+    struct ChainInfo {
+        // Price of chain's gas token in USD, scaled to wei
+        uint128 gasTokenPrice;
+        // Price of chain's 1 gas unit in wei
+        uint128 gasUnitPrice;
+    }
+
+    /// @dev Ratio between src and dst gas price ratio.
+    /// Used for calculating a fee for sending a msg from src to dst chain.
+    /// Updated whenever "gas information" is changed for either source or destination chain.
+    struct ChainRatios {
+        // USD price ratio of dstGasToken / srcGasToken, scaled to wei
+        uint96 gasTokenPriceRatio;
+        // How much 1 gas unit on dst chain is worth,
+        // expressed in src chain wei, multiplied by 10**18 (aka in attoWei = 10^-18 wei)
+        uint160 gasUnitPriceRatio;
+        // To calculate gas cost of tx on dst chain, which consumes gasAmount gas units:
+        // (gasAmount * gasUnitPriceRatio) / 10**18
+        // This number is expressed in src chain wei
+    }
+
+    /*┌──────────────────────────────────────────────────────────────────────┐
+      │                                EVENTS                                │
+      └──────────────────────────────────────────────────────────────────────┘*/
+
+    event ChainInfoUpdated(uint256 indexed chainId, uint256 gasTokenPrice, uint256 gasUnitPrice);
+
+    event MarkupsUpdated(uint256 markupGasDrop, uint256 markupGasUsage);
```

Comment:
What's a markup (might want to be a little more verbose here)

Reply:
@trajan0x more clear docs are added in [73d8273](https://github.com/synapsecns/synapse-contracts/pull/158/commits/73d82735afa7000ff925594ebcd3803569536e58)